src/zenml/integrations/kubeflow/docker_utils.py

```diff
@@ -283,6 +286,20 @@ def get_image_digest(image_name: str) -> Optional[str]:
         return None
 
 
+def is_local_image(image_name: str) -> bool:
+    """Returns whether an image was pulled from a registry or not."""
+    docker_client = DockerClient.from_env()
+    images = docker_client.images.list(name=image_name)
+    if images:
+        # An image with this name is available locally -> now check whether it
+        # was pulled from a repo or built locally (in which case the repo
+        # digest is empty)
+        return get_image_digest(image_name) is None
```

Comment:
Very nice ! You have no idea how much this has been bugging me... or maybe you do :)

Reply:
Trust me, I do .. :)